Detectors/Base/include/DetectorsBase/BaseHits.h

```diff
@@ -0,0 +1,67 @@
+#ifndef ALICEO2_BASE_HIT_H
+#define ALICEO2_BASE_HIT_H
+#include "FairMultiLinkedData_Interface.h"
+#include "Math/GenVector/DisplacementVector3D.h"
+#include "Math/GenVector/PositionVector3D.h"
+#include "Math/SVector.h"
+
+template <typename T>
+using Point3D = ROOT::Math::PositionVector3D<ROOT::Math::Cartesian3D<T>, ROOT::Math::DefaultCoordinateSystemTag>;
+template <typename T>
+using Vector3D = ROOT::Math::DisplacementVector3D<ROOT::Math::Cartesian3D<T>, ROOT::Math::DefaultCoordinateSystemTag>;
+
+namespace AliceO2
+{
+namespace Base
+{
+// a set of configurable classes to define basic hit types
+// these are meant to be an alternative to FairMCPoint
+// which always includes the momentum and is purely based on double values
+
+// A basic hit class encoding cartesian coordinates and energy loss
+// we need to derive from FairMultiLinkedData_Interface for the
+// MC truth handling
+template <typename T>
+class BasicXYZEHit : public FairMultiLinkedData_Interface
+{
+  int mDetectorID; // the detector/sensor id
+  int mTrackID;    // this might be part of link
+  T mTime;         // time of flight
+  Point3D<T> mPos; // cartesian position of Hit
+  T mELoss;
+
```

Comment:
I think at least the mEloss and mTime can be fixed to float, in any case, their precision and operations on them are not related to that of mPos.
Also mDetectorID can be fixed to short (sorry for telling you opposite before, detectors will have internal enumeration of their sensors and we will have < 25k sensors).

Cheers,
Ruben